pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1534,6 +1538,12 @@ def _handle_distribution_tree(declarative_content):
                 for update_reference in update_references:
                     update_reference.update_record = update_record
                     update_references_to_save.append(update_reference)
+            elif isinstance(declarative_content.content, Package):
+                artifact = declarative_content.d_artifacts[0]  # Packages have 1 artifact
+                signer_key_id = parse_signer_id(artifact.artifact.file.path)
```

Comment:
Just checking, is there any reason for this to be in `_post_save()` of this particular stage?  Or does it just need to happen at some point after artifact download?

Reply:
One issue, and maybe this is why you did it this way:

Lets say you sync the same repo twice, once "on_demand" into one repo without the key check and once "immediate" into a second repo with the key check.  The "on_demand" sync cannot populate the gpg key field on the content unit because it doesn't have access to the content, and the "immediate" sync would (at the current moment) try to re-use the existing content unit because the natural key matches up.

So the standard ContentSaver stage would ignore those content units because they're already "saved" despite needing to be updated.  Hence putting it in `_post_save()` so that it's not ignored.  Is that right?

Reply:
I don't know anything about "on_demand" syncing. I put it in `_post_save()` because it was an already-overridden method, in a stage late enough that I thought it was safe to assume that the artifact was actually downloaded, next to checks that were somewhat-similar.

If there's a better place for it I'm relying on you all to tell me where to put it.

Reply:
I think the `pre_save` hook would be better because the content unit doesn't need to be re-saved.  Whereas doing it aftewards results in an update after-the-fact.